### PR TITLE
Support `@JsonIgnoreProperties(allowGetters,allowSetters)`, align with Jackson behaviour

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-serialization/issues/581
Fixes https://github.com/micronaut-projects/micronaut-serialization/issues/550

### CODE REVIEW

File: serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java [hunk 1]
```diff
@@ -687,6 +686,16 @@ private void visitSubtype(ClassElement supertype, ClassElement subtype, VisitorC
             return;
         }
 
+        if (!subtype.hasAnnotation(SerdeConfig.SerIgnored.class)) {
+            // Replicate the Jackson behaviour of using the ignore annotation from supertype but allow to override it
+            AnnotationValue<SerdeConfig.SerIgnored> serIgnored = supertype.getAnnotation(SerdeConfig.SerIgnored.class);
```
Comment (yawkat) [inline serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: what if it's on the supersupertype

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java [hunk 2]
```diff
@@ -175,12 +175,12 @@ Entry getEntry(int i) throws IOException {
             if (buffer.size() > i) {
                 return buffer.get(i);
             } else {
-                if (buffer.size() != i) {
```
Comment (yawkat) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java:178]: why is this change necessary? is something calling decodeKey again after it already returned null?

File: serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java (hunk 1 above)
Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: The annotation is not inherited

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java (hunk 2 above)
Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/DemuxingObjectDecoder.java]: Reverted

File: serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java (hunk 1 above)
Comment (yawkat) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: in a hierarchy of A extends B and B extends C, and C has @SerIgnored, I think this will mean that A will only get SerIgnored if B is visited before A?

Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: No, `@SerIgnored` is not inherited, so adding it to `B` shouldn't add it to `A`.

Comment (yawkat) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: but youre copying it here? i dont understand what youre saying. this code copies the annotation from the supertype, but not if it's further up, even though i assume in jackson it would still be considered for such a hierarchy

Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: Well if the annotation is not inherited, putting it on B shouldn’t be present on A etc.

Comment (yawkat) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: but with this code, putting it on B does copy it to A.

Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: The subtypes in this code are the subtypes of the base class defined by JsonSubTypes. The logic is called one. It’s not called on every type.

Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: See https://github.com/micronaut-projects/micronaut-serialization/pull/640/files#diff-37b66da15700d594aac6c8e6eab4763e00f8dc3b0d769186e203b400a7a557b7R800

The ignore is not inherited. With subtypes it acts the same, if you put ignore again and want to ignore something from the supertype you need to repeat it.

Comment (yawkat) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: ive pushed a pendingfeature with the setup that i mean. the test passes with jackson-databind.

Comment (dstepanov) [reply serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java:691]: Ahh, looks like Jackson respects the last definition. I can take a look in the next PR

**Review by yawkat — APPROVED** [review]
